### PR TITLE
[NUI] Add Dialog and AlertDialog classes

### PR DESCRIPTION
**Pull request by jaehyun0cho**
### Description of Change ###

Dialog is used to show a popup content with background scrim.
Navigator provides static method ShowDialog which pushes Page containing
Dialog.

AlertDialog is used to show a dialog with title, message and action
buttons.
Navigator provides static method ShowAlertDialog which pushes Page
containing AlertDialog.

### API Changes ###

 - ACR:

### CODE REVIEW

**Review by AdunFang — APPROVED** [review]
It is OK.

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs [hunk 1]
```diff
@@ -335,5 +335,94 @@ public static Navigator GetDefaultNavigator(Window window)
...
+
+        /// <summary>
+        /// Shows a dialog by pushing a page containing dialog to default navigator.
+        /// </summary>
+        /// <param name="content">The content of Dialog.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void ShowDialog(View content = null)
+        {
+            var window = NUIApplication.GetDefaultWindow();
+            var defaultNavigator = window.GetDefaultNavigator();
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs:347]: I have question that is not directly related with this patch.
Does the method name `GetDefaultNavigator` means there are multiple Navigator in a window?

**Comment by dongsug-song** [comment]
@jaehyun0cho 
Hello,
could you check the @rabbitfor 's reveiw?
thank you.

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs (hunk 1 above)
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs:347]: Yes, it is possible if a user creates multiple Navigator instances in a window.
e.g.
```
var window = NUIApplication.GetDefaultWindow();

var navigator = new Navigator();
window.Add(navigator);

var navigator2 = new Navigator();
window.Add(navigator2);
```

If window.GetDefaultNavigator() is called for the first time, then Navigator instance is created and stored in Dictionary with the window.
If window.GetDefaultNavigator() is called again, then the stored Navigator instance is returned.

**Review by Seoyeon2Kim — APPROVED** [review]
Looks good to me, but other's approval is needed.

**Comment by dongsug-song** [comment]
@jaehyun0cho 
Hello,
could you check the @rabbitfor 's reveiw?
thank you.
